libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m

```diff
@@ -69,14 +84,36 @@ - (void)setUp
 - (void)tearDown
 {
     // Tear-down code here.
+    [self cleanup];
     [[SFRestAPI sharedInstance] cleanup];
     [NSThread sleepForTimeInterval:0.1];  // Some test runs were failing, saying the run didn't complete.  This seems to fix that.
     [super tearDown];
 }
 
-#pragma mark - help methods
 
 
+#pragma mark - helper methods
+
+// Helper method to delete any entities created by one of the test
+- (void) cleanup {
```

Comment:
The same thing we do on Android.
Tests always call generateRecordName to name their records and at tearDown, we do a sosl query to get all the records that were created and then we delete them.
That way tests do not need to worry about cleaning up.